### PR TITLE
Mustafa/week2

### PR DESCRIPTION
**Pull request by mayqf**

### CODE REVIEW

File: Week2/homework2/queries.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const Query3 = `
+SELECT COUNT(*) as Number_Of_Cities 
+FROM city WHERE CountryCode 
+IN(SELECT CountryCode FROM countryLanguage WHERE Language = ?) `;
+
+const Query4 = `SELECT Name as Countries 
+FROM country WHERE region = ? and Code 
+IN(SELECT CountryCode FROM countryLanguage WHERE Language = ? and IsOfficial = 'T')`;
+
+const Query5 = `SELECT country.Continent,COUNT(distinct Language) as Number_Of_Spoken_Languages
```
Comment (mogaleaf) [inline Week2/homework2/queries.js]: This query is not working :)

File: Week2/homework2/queryfunctions.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+};
+const queryCities = async () => {
+  const { languageName } = await inquirer.prompt(input3);
+  const result = await execQuery(Query3, languageName);
+  console.log(`The language of ${languageName} is spoken in ${result[0].Number_Of_Cities} cities`);
+};
+const queryOfficiallanguages = async () => {
+  const { regionName, languageName } = await inquirer.prompt(input4);
+  const result = await execQuery(Query4, [regionName, languageName]);
+  const answer = result.length === 0 ? 'FALSE' : result.map(item => Object.values(item).toString());
```
Comment (mogaleaf) [inline Week2/homework2/queryfunctions.js]: yes it's good like that

File: Week2/homework2/queryfunctions.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+3. Find the number of cities in which language Z is spoken 
+4. Accept the region and language from the user.Are there any countries in 
+   this region with the given language as the official language ?  
+5. List all the continents with the number of languages spoken in each continent\n`;
+async function seedDatabase() {
+  try {
+    con.connect();
+
+    console.log(explanation);
+    const { choice } = await inquirer.prompt(input5);
```
Comment (mogaleaf) [inline Week2/homework2/queryfunctions.js]: I love the arrow choice :) The program doesn't end when I select, anyway to make it end or give me back the menu ?

File: Week2/homework2/part2trigger.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
+DELIMITER //
+CREATE TRIGGER mustafa_trigger
+    BEFORE INSERT
+        ON countrylanguage
+            FOR EACH ROW
+            BEGIN
+                DECLARE message VARCHAR(100);
+                DECLARE number0fLanguages INT;
+                SELECT COUNT(Language) INTO numberOfLanguages FROM countrylanguage WHERE country.countrycode=new.countrycode;
```
Comment (mogaleaf) [inline Week2/homework2/part2trigger.sql]: small typo here between number0fLanguages and  numberOfLanguages

File: Week2/homework2/queries.js (hunk 1 above)
Comment (mayqf) [reply Week2/homework2/queries.js]: many thanks for your feedback.I will try to take care:)

Comment (mayqf) [reply Week2/homework2/queries.js]: So you Query5 does not work??

File: Week2/homework2/part2trigger.sql (hunk 4 above)
Comment (mayqf) [reply Week2/homework2/part2trigger.sql]: Ok:)

File: Week2/homework2/queryfunctions.js (hunk 3 above)
Comment (mayqf) [reply Week2/homework2/queryfunctions.js]: Actually, I wanted to do it in a while loop. I wanted to execute program(asking queries/menu) until user select quit. But I could not manage it. 
Yes, you are right. After making a selection and allocating proper inputs, the program doe not quit. How can I solve it?

**Comment by mayqf** [comment]
m

**Comment by mayqf** [comment]
n

File: Week2/homework2/queryfunctions.js (hunk 3 above)
Comment (mogaleaf) [reply Week2/homework2/queryfunctions.js]: Why are you executing your queryCapital func into the executeQuery statement ? executeQuery that you are using correctly after is supposed to be used with the sql statement, not a func. Try to remove the executeQuery and just await your queryCapital func .

Comment (mayqf) [reply Week2/homework2/queryfunctions.js]: yes you are right. thanks

**Comment by mayqf** [comment]
rvs

**Comment by mayqf** [comment]
s

**Comment by mayqf** [comment]
s

**Pull request closed without merging**